drivers/spi/spi_mchp_mec5_qspi.c

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright (c) 2025 Microchip Technology Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT microchip_mec5_qspi
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_mec5, CONFIG_SPI_LOG_LEVEL);
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/sys/util.h>
+#include <soc.h>
+#include <zephyr/irq.h>
+
+#include "spi_context.h"
+
+/* MEC5 HAL */
+#include <device_mec5.h>
+#include <mec_ecia_api.h>
+#include <mec_espi_taf.h>
+#include <mec_qspi_api.h>
+
+struct mec5_spi_devices {
+	uint32_t cs_timing;
+	uint8_t cs;
+	uint8_t sck_tap;
+	uint8_t ctrl_tap;
+	uint8_t flags;
+};
+
+/* Device constant configuration parameters */
+struct mec5_qspi_config {
+	struct mec_qspi_regs *regs;
+	int clock_freq;
+	const struct pinctrl_dev_config *pcfg;
+	void (*irq_config_func)(void);
+	uint8_t ovrc;
+	uint8_t num_child_devices;
+	const struct mec5_spi_devices *child_devices;
+};
+
+#define MEC5_QSPI_XFR_FLAG_START BIT(0)
+#define MEC5_QSPI_XFR_FLAG_BUSY  BIT(1)
+#define MEC5_QSPI_XFR_FLAG_LDMA  BIT(2)
+
+/* Device run time data */
+struct mec5_qspi_data {
+	struct spi_context ctx;
+	const struct spi_buf *rxb;
+	const struct spi_buf *txb;
+	size_t rxcnt;
+	size_t txcnt;
+	volatile uint32_t qstatus;
+	volatile uint32_t xfr_flags;
+	size_t total_tx_size;
+	size_t total_rx_size;
+	size_t chunk_size;
+	uint32_t rxdb;
+	uint32_t byte_time_ns;
+	uint32_t freq;
+	uint32_t operation;
+	uint8_t cs;
+};
+
+static const enum mec_qspi_signal_mode mec5_qspi_sig_mode[4] = {
+	MEC_SPI_SIGNAL_MODE_0, MEC_SPI_SIGNAL_MODE_1, MEC_SPI_SIGNAL_MODE_2, MEC_SPI_SIGNAL_MODE_3};
+
+static int spi_feature_support(const struct spi_config *config)
+{
+	/* NOTE: bit(11) is Half-duplex(3-wire) */
+	if (config->operation & (SPI_TRANSFER_LSB | SPI_OP_MODE_SLAVE | SPI_MODE_LOOP | BIT(11))) {
+		LOG_ERR("Driver does not support LSB first, slave, loop back, or half-duplex");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_CS_ACTIVE_HIGH) {
+		LOG_ERR("CS active high not supported");
+		return -ENOTSUP;
+	}
+
+	if (SPI_WORD_SIZE_GET(config->operation) != 8) {
+		LOG_ERR("Word size != 8 not supported");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+int get_cs_timing_from_dt(const struct device *dev, uint8_t cs, uint32_t *cstm)
+{
+	const struct mec5_qspi_config *devcfg = dev->config;
+
+	if (!cstm) {
```

Comment:
No implicit boolean checks (also for other functions)

```suggestion
	if (cstm == NULL) {
```